### PR TITLE
Hotfix/cnb

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Fixes NPE with chisel and bits
- Fixes bug where too many items get sent to the worker.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java [hunk 1]
```diff
@@ -294,7 +294,10 @@ public int getTotalRequiredAmount(final ItemStack deliveredItemStack)
             final int hashCode = deliveredItemStack.hasTagCompound() ? deliveredItemStack.getTagCompound().hashCode() : 0;
             final BuildingBuilderResource resource
                     = ((BuildingBuilder) getOwnBuilding()).getNeededResources().get(deliveredItemStack.getUnlocalizedName() + ":" + deliveredItemStack.getItemDamage() + "-" + hashCode);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java:296]: MAJOR Split this 185 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java [hunk 2]
```diff
@@ -752,7 +752,7 @@ private void fulfill(@NotNull final Button button)
             {
                 itemStack = inventory.getStackInSlot(InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(inventory), requestPredicate));
             }
-            MineColonies.getNetwork().sendToServer(new TransferItemsToCitizenRequestMessage(citizen, itemStack, isCreative ? amount : count, citizen.getColonyId()));
+            MineColonies.getNetwork().sendToServer(new TransferItemsToCitizenRequestMessage(citizen, itemStack, isCreative ? amount : Math.min(amount, count), citizen.getColonyId()));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java]: MAJOR Split this 183 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 61 issues
* ![MAJOR][MAJOR] 58 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L364](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L364): Refactor this method to reduce its Cognitive Complexity from 23 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L373](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L373): Reduce this anonymous class number of lines from 65 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L395](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L395): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L396](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L396): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L599](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L599): Split this 183 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/465501a03921dd64ff6f0e162b9194ffa9118e6f/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — DISMISSED** [review]

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 60 issues
* ![MAJOR][MAJOR] 57 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L364](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L364): Refactor this method to reduce its Cognitive Complexity from 23 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L373](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L373): Reduce this anonymous class number of lines from 65 to at most 40, or make it a named class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L395](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L395): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L396](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L396): Assign this magic number 10 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [WindowCitizen.java#L599](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java#L599): Split this 183 characters long line (which is greater than 180 authorized). [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/1089a0092a8becd3962068c654a782e4a33b267f/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**